book/tutorials/gpr/get_gpr_and_density.ipynb

```diff
@@ -0,0 +1,464 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Retrieve GPR data centered on a pit\n",
+    "\n",
+    "**Goal**: Visualize a set of GPR snow depths around a pit and calculate the avg density from the pit\n",
+    "\n",
+    "**Approach**: \n",
+    "\n",
+    "1. Retrieve the pit location from the Layer Data table \n",
+    "2. Build a circle around the pit location \n",
+    "3. Request all the GPR data with a distance of our pit\n",
+    "4. Request the Pit Data to get density layers and calculate the avg\n",
+    "4. Plot depths!\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Process\n",
+    "### Step 1: Get the pit/site coordinates"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Import our DB access function\n",
+    "from snowexsql.db import get_db\n",
+    "\n",
+    "# Import the two tables we need GPR ---> PointData, Density (Pits) --> LayerData \n",
+    "from snowexsql.data import PointData, LayerData\n",
+    "from snowexsql.conversions import query_to_geopandas\n",
+    "\n",
+    "# Import to make use of the posgis functions on the db that are not necessarily in python \n",
+    "from sqlalchemy import func, Float\n",
+    "# Import datetime module to filter by a date\n",
+    "import datetime \n",
+    "\n",
+    "\n",
+    "# use numpy to calculate the average of the density results \n",
+    "import numpy as np \n",
+    "\n",
+    "# Import matplotlib\n",
+    "import matplotlib.pyplot as plt\n",
```

Comment:
if you add a line here `%config InlineBackend.figure_format='retina'` your figure text will come out cleaner